Updating docs

mcottontensor opened the pull request.

mcottontensor committed:
Brought back the signalling messages docs. Updated for new protocol additions.

mcottontensor committed:
Added protocolVersion to config message docs.

lukehb reviewed (changes requested):
As mentioned, lets:

Add a link to that doc in the /Signalling and /SignallingWebServer readme.md to the existing protocol doc.
Add a link to the sequence diagram in the /Signalling readme.md

mcottontensor committed:
Rebuilt docs for proto, Common and Signalling.
Added new Protocol.md in Signalling/docs that is a snip from the old
message documentation that covers terms and signalling protocol
negotiation.

mcottontensor committed:
Adding further documentation section

mcottontensor committed:
Removed irrelevant table of contents.

mcottontensor committed:
Removed unmaintainable version section.

mcottontensor committed:
Fixing formatting of further documentation section.

mcottontensor committed:
Fixing link for protocol messages.

mcottontensor committed:
Adding message doc link to signalling readme.

lukehb reviewed (approved):
lgtm
